### PR TITLE
add fcntl for ql_file

### PR DESCRIPTION
**Pull request by cq674350529**
try to fix https://github.com/qilingframework/qiling/issues/642

### CODE REVIEW

**Comment by cq674350529** [comment]
The failed test seems to be related to https://github.com/qilingframework/qiling/issues/640

File: qiling/os/filestruct.py [hunk 1]
```diff
@@ -50,6 +48,12 @@ def close(self):
     def fstat(self):
         return Fstat(self.__fd)
     
+    def fcntl(self, fcntl_cmd, fcntl_arg):
+        try:
+            return fcntl.fcntl(self.__fd, fcntl_cmd, fcntl_arg)
+        except Exception:
+            pass
+
```
Comment (xwings) [inline qiling/os/filestruct.py:56]: why try and except. anything we need to worry about ?

Comment (cq674350529) [reply qiling/os/filestruct.py:56]: There is a chance that the module `fcntl` may be imported unsuccessfully, as shown in the begin of the file. So maybe this `try-except` is used for this?

Comment (xwings) [reply qiling/os/filestruct.py:56]: i guess is smart for us to remove all of this
i mean try import above and try import here.

Comment (cq674350529) [reply qiling/os/filestruct.py:50]: I try to remove both `try-except` catch, and test failed on windows for no module `fnctl`. There is another `fcntl.ioctl()` in `qiling/os/filestruct.py`, I'm not sure if it's used ot not.  So I try another way: only call `fcntl()` on ql_socket, which is submitted in another pull request.

**Comment by xwings** [comment]
let me take a look and see how can we fix this mess :)

**Pull request closed without merging**